src/Pointer.ts

```diff
@@ -3,25 +3,36 @@ import { makeLine } from "./Utils";
 import { Node } from "./Node";
 import { Variable } from "./Variable";
 import { Point } from "./Types";
+import Config = require("./Config");
 
 export class Pointer {
     private origin: Node | Variable;
     private destination: Node;
-
+    private canvas: fabric.Canvas;
     private line: fabric.Line;
+    private selfLoop: fabric.Path;
     private arrowhead: fabric.Line[];
 
     constructor(origin: Node | Variable, canvas: fabric.Canvas) {
         this.origin = origin;
         this.destination = null;
-
+        this.canvas = canvas;
         this.line = makeLine();
+        this.selfLoop = new fabric.Path("",{ fill: '', stroke: 'black', objectCaching: false, strokeWidth: 2 });
         this.arrowhead = [makeLine(), makeLine()];
         canvas.add(this.line, ...this.arrowhead);
     }
 
     public set(nodePointedTo: Node): void {
         this.destination = nodePointedTo;
+
+        this.canvas.remove(this.line, this.selfLoop, ...this.arrowhead);
```

Comment:
I think you could call `this.erase()` here to avoid code duplication.

Reply:
duh of course. I meant to do that...